services/blockchain-indexer/shared/utils/transactionsUtils.js

```diff
@@ -39,6 +53,53 @@ const getTxnMinFee = async (
 	},
 );
 
+const normalizeTransaction = async tx => {
+	const metadata = await getSystemMetadata();
+	const filteredModule = metadata.modules.find(module => module.id === tx.moduleID);
+	const filteredCommand = filteredModule.commands.find(s => s.id === tx.commandID);
+
+	const { params } = tx;
+	const decodedParams = codec.decode(filteredCommand.params, Buffer.from(params, 'hex'));
+	const parsedTxParams = parseInputBySchema(decodedParams, filteredCommand.params);
+
+	const schema = await requestConnector('getSchema');
+	const parsedTx = parseInputBySchema(tx, schema.transaction);
+	const parsedTxWithParams = { ...parsedTx, params: parsedTxParams };
+
+	const txBuffer = codec.encode(
+		schema.transaction,
+		{ ...parsedTx, params: Buffer.from(params, 'hex') },
+	);
+
+	// TODO: Remove when transaction ID is available from SDK
+	tx.id = hash(txBuffer).toString('hex');
+	tx.minFee = await getTxnMinFee(parsedTxWithParams);
+	tx.size = txBuffer.length;
+	tx.params = decodedParams;
+
+	const normalizedTransaction = parseToJSONCompatObj(tx);
+
+	const availableModuleCommands = await getAvailableModuleCommands();
+	const [{ id, name }] = availableModuleCommands
+		.filter(module => module.id === String(tx.moduleID).concat(':', tx.commandID));
```

Comment:
Is it necessary to parse the transaction here? Could we not parse it directly in the return statement?
```suggestion
	const normalizedTransaction = parseToJSONCompatObj(tx);

	const availableModuleCommands = await getAvailableModuleCommands();
	const { id, name } = availableModuleCommands
		.find(module => module.id === String(tx.moduleID).concat(':', tx.commandID));
```